tests/pytest_translate

```diff
@@ -0,0 +1,131 @@
+
+args = parser.parse_args()
+
+
+def eprint(*args, **kwargs):
+    print(*args, file=sys.stderr, **kwargs)
+
+
+eprint(f"input_file: {args.input_file}")
+eprint(f"output_file: {args.output_file}")
+
+with open(args.input_file, "r") as file:
+    input_source = file.read()
+
+try:
+    input_ast = ast.parse(input_source, args.input_file)
+except SyntaxError as syntaxError:
+    sys.exit(f"ERROR: failed to parse '{args.input_file}': {syntaxError}")
+
+
+class Language(enum.Enum):
+    PYTHON = 0
+    CSHARP = 1
+
+
+def get_fileext_language(filename):
+    if filename.endswith(".py"):
+        return Language.PYTHON
+    if filename.endswith(".cs"):
+        return Language.CSHARP
+    return None
+
+
+output_language = get_fileext_language(args.output_file)
+if not output_language:
+    sys.exit(f"ERROR: unknown file extension from '{args.output_file}'")
+eprint(f"output_language: {output_language}")
+
+
+class IrGenVisitor(ast.NodeVisitor):
+
+    def __init__(self):
+        self.ast_depth = 0
+        self.current_scope = None
+        self.scope_depth = 0
+
+    def log(self, msg):
+        print(f"{self.ast_depth*' '}{msg}")
+
+    def visitimpl(self, node):
+        self.ast_depth += 1
+        ast.NodeVisitor.generic_visit(self, node)
+        self.ast_depth -= 1
+
+    def visit_new_scope(self, node, new_scope):
+        parent_scope = self.current_scope
+        self.current_scope = new_scope
+        self.scope_depth += 1
+        self.visitimpl(node)
+        assert(new_scope == self.current_scope)
+        self.scope_depth -= 1
+        self.current_scope = parent_scope
+
+    def generic_visit(self, node):
+        self.log(f"GENERIC: {type(node).__name__}")
+        self.visitimpl(node)
+
+    def visit_Module(self, node):
+        assert(self.ast_depth == 0)
+        assert(self.current_scope is None)
+        assert(self.scope_depth == 0)
+        module = pythonir.Module()
+        self.visit_new_scope(node, module)
+        return module
+
+    def visit_Import(self, node):
+        assert(self.current_scope)
+        self.log(f"visit_Import names={node.names}")
+        self.current_scope.add_import(node.names)
+
+    def visit_ImportFrom(self, node):
+        assert(self.current_scope)
+        self.log(f"visit_ImportFrom names={node.names}")
+        self.current_scope.add_import_from(node.module, node.names)
+
+    def visit_FunctionDef(self, node):
+        self.log(f"visit_FunctionDef name={node.name} args={node.args}")
+        assert(self.current_scope)
+        assert(self.scope_depth > 0)
+        parent_scope = self.current_scope
+        # args, body, name, returns
+        func = pythonir.Function(node.name)
+        self.visit_new_scope(node, func)
+        self.current_scope.add_function_def(func)
+
+    def visit_Assign(self, node):
+        assert(self.current_scope)
+        self.log(f"visitAssign targets={node.targets} value={node.value}")
+        # self.current_scope.add_assign(node)
```

Comment:
?

Reply:
something that will need to be implemented later, I commented it out for now so I didn't develop too much code without review